library/Fission/Mail/SendInBlue/Types.hs

```diff
@@ -0,0 +1,53 @@
+module Fission.Mail.SendInBlue.Types 
+  ( SendEmailReq(..)
+  , CustomParams(..)
+  , SendEmailRes(..)
+  , module Fission.Mail.SendInBlue.ApiKey.Types
+  , module Fission.Mail.SendInBlue.TemplateId.Types
+  ) where
+
+import           Fission.Prelude
+import           Fission.Mail.Recipient.Types
+import           Fission.Mail.SendInBlue.ApiKey.Types
+import           Fission.Mail.SendInBlue.TemplateId.Types
+
+import           Fission.User.Username.Types
+import           Servant.Client
+
+import qualified Data.Vector as Vector
+
+
+data SendEmailReq = SendEmailReq
+  { templateId :: !TemplateId
+  , to :: ![Recipient]
+  , params :: !CustomParams
+  } 
+
+instance ToJSON SendEmailReq where
+  toJSON SendEmailReq { templateId = templateId', to = to', params = params' } =
+    Object [ ("templateId", Number $ fromIntegral (unTemplateId templateId'))
+           , ("to", Array $ Vector.fromList (toJSON <$> to'))
+           , ("params", toJSON params')
+           ]
+
+data CustomParams = CustomParams
+  { verifyLink :: !BaseUrl 
+  , username   :: !Username 
+  } 
+
+instance ToJSON CustomParams where
+  toJSON CustomParams { verifyLink = verifyLink', username = username' } =
+    Object [ ("VERIFY_LINK", toJSON verifyLink' )
+           , ("USERNAME", toJSON username' )
+           ]
+
+
+data SendEmailRes = SendEmailRes
+  { messageId :: Text }
```

Comment:
1. Since this is a single parameter, this should be a `newtype`
2. `deriving newtype (Eq, Show, Display)`